### PR TITLE
Add Enum.filter/2 lesson (#1342)

### PR DESCRIPTION
**Pull request by wilbertliu**
Hello folks 👋🏻,

This PR is the real work based on discussion from #1342.
Any thoughts?

Thanks! 🙏🏻

### CODE REVIEW

File: en/lessons/basics/enum.md [hunk 1]
```diff
@@ -143,6 +143,15 @@ Enum.max([], fn -> :bar end)
 :bar
 ```
 
+### filter
+
+`filter/2` returns collection's elements that match the condition from the given function.
```
Comment (doomspork) [inline en/lessons/basics/enum.md]: How about something like:

> The `filter/2` function enables us to filter the collection to include only those elements that evaluate to `true` using the provided function.

Comment (wilbertliu) [reply en/lessons/basics/enum.md]: Sounds clearer Sean! 😀 Will update then!

Comment (nscyclone) [reply en/lessons/basics/enum.md]: Thanks, @wilbertliu!
Should we bump the minor version of the lesson?

Comment (wilbertliu) [reply en/lessons/basics/enum.md]: Hello @nscyclone, I'm not sure I understand what you intended to ask. Could you please elaborate more? 🙏🏻

Comment (nscyclone) [reply en/lessons/basics/enum.md]: @wilbertliu, there is a variable called [`version`](https://github.com/elixirschool/elixirschool/pull/1349/files#diff-dfa183a1580b77baec65b2a1f09e3948L2) in the [front matter](https://jekyllrb.com/docs/frontmatter/) of every lesson/translation that helps to keep an eye on differences between original lessons and their translations. Usually when we introduce changes to an existing file we also increase `major`, `minor` or `patch` version. You can find a little more information about this in [`README`](https://github.com/elixirschool/elixirschool#translated-lesson).

In my opinion, in this case update of the `minor` version number would fit well to reflect the incoming changes, WDYT?

Comment (wilbertliu) [reply en/lessons/basics/enum.md]: @nscyclone Ah, I see. And yes, this should be `minor`. Who is in charge to change that though? Me? Thanks a lot for explaining! 🙂

Comment (nscyclone) [reply en/lessons/basics/enum.md]: @wilbertliu Sorry for a late reply! Would be great if you'd take care of that on your own. Thanks in advance!

Comment (wilbertliu) [reply en/lessons/basics/enum.md]: @nscyclone No problem mate! I think I'm done? 😊

**Comment by wilbertliu** [comment]
Hello @doomspork 🙂

I know it's already merged, but I couldn't see the lesson on the production, which is https://elixirschool.com/en/lessons/basics/enum/

Is that expected? But I could see my photo at the `Contributors` section.

Thanks! 🙏🏻

**Comment by doomspork** [comment]
@wilbertliu there was an issue with Heroku deployments I have to look into today.

**Comment by nscyclone** [comment]
@wilbertliu Glad to see it [live](https://elixirschool.com/en/lessons/basics/enum/#filter)! Thanks once again!

**Pull request merged**